test/registration_codes_loader_spec.rb

```diff
@@ -0,0 +1,133 @@
+#! /usr/bin/env rspec
+
+require_relative "spec_helper"
+
+describe Registration::RegistrationCodesLoader do
+  class RegistrationCodesLoaderTest
+    include Registration::RegistrationCodesLoader
+  end
+
+  subject { RegistrationCodesLoaderTest.new }
+
+  describe "#reg_codes_from_usb_stick" do
+    let(:tempfile) { "/whatever1" }
+    let(:xmlname)  { "regcodes.xml" }
+    let(:txtname)  { "regcodes.txt" }
+
+    before do
+      expect(subject).to receive(:with_tempfile) do |_pattern, &block|
+        block.call(tempfile)
+      end
+    end
+
+    context "when only regcodes.xml exists" do
+      it "delegates to reg_codes_from_xml" do
+        expect(subject).to receive(:get_file_from_url)
+          .with(scheme: "usb", host: "", urlpath: "/#{xmlname}",
+                               localfile: tempfile,
+                               urltok: {}, destdir: "")
+          .and_return(true)
+
+        expect(subject).to receive(:reg_codes_from_xml)
+          .with(tempfile)
+          .and_return("a" => "b")
+
+        expect(subject.reg_codes_from_usb_stick).to eq("a" => "b")
+      end
+    end
+
+    context "when only regcodes.txt exists" do
```

Comment:
And what if both exist? Which one is preferred? I think this case should be also covered so we still prefer the same. (And to detect a regression when the preferred file is changed.)